app/services/page_archiver.rb

```diff
@@ -0,0 +1,23 @@
+class PageArchiver
+  include Rails.application.routes.url_helpers
+
+  def self.archive!(page)
+    new(page).archive!
+  end
+
+  def self.unarchive!(page)
+    new(page).unarchive!
+  end
+
+  def initialize(page)
+    @page = page
+  end
+
+  def archive!
+    @page.update! publish_status: :archived
```

Comment:
ah! didn't know that. Now that you mention that, it pretty much doesn't make any sense to have this service. I originally thought I'd be pushing a job from here. But now I'd call `@page.archive!` from the controller directly.